src/settingsNames.ts

```diff
@@ -1,6 +1,11 @@
 const settingsNames = {
   authType: 'authType',
   autoOpenLinksInBrowser: 'autoOpenLinksInBrowser',
+  clientId: 'clientId',
+  clientSecret: 'clientSecret',
+  clientCertificateFile: 'clientCertificateFile',
+  clientCertificateBase64Encoded: 'clientCertificateBase64Encoded',
+  clientCertificatePassword: 'clientCertificatePassword',
```

Comment:
Say we let you configure an existing app that uses a cert or secret. If you only specify the app ID it wouldn't be working. So IMHO we should either allow you to configure all necessary pieces to use the app or none. We could choose to only allow configuring public client apps and then defer to env vars for secrets, but that would lead to a broken setup experience that works for some things but not for others.